# Comparison with Meritt, _The Athenian Year_

In [1]:
import heniautos as ha
ha.init_data()

'de422.bsp'

In [2]:
meritt_intercalations = list("IOOIOIOIOOOIOOIOIOIIOOOOIOOIO")

meritt_hek1 = ((6, 28), (7, 17), (7, 7), (6, 26), (7, 14), (7, 3), (7, 22), (7, 11), (7, 29), (7, 19), (7, 9), 
               (6, 28), (7, 16), (7, 5), (7, 24), (7, 13), (7, 1), (7, 20), (7, 10), (7, 29), (8, 16), (8, 5), 
               (7, 25), (7, 14), (7, 3), (7, 22), (7, 11), (6, 30), (7, 18))

meritt_len = (384, 355, 354, 384, 354, 384, 354, 384, 355, 355, 354, 384, 354, 354, 384, 354, 384, 355, 
              384, 384, 354, 354, 354, 355, 384, 354, 354, 384, 355)

def month(m):
    if m == 6:
        return "Jun"
    if m == 7:
        return "Jul"
    return "Aug"

Meritt (1961) 218 presents a summary of his calculations for the years 433/2-405/4 BCE. Reproduced below are his results for whether the year was intercalated (I) or not (O), the Julian date of Hekatombaiṓn 1 and the length of the year

In [3]:
print("Year    | Int? | Hek. 1 | Length")
print("-"*32)
for i, year in enumerate(reversed(range(405, 434))):
    if not year % 5:
        print("-"*32) 
    print(f"{year}/{year-1} | {meritt_intercalations[i]:^4} | "
         f"{month(meritt_hek1[i][0])}-{meritt_hek1[i][1]:02} | {meritt_len[i]:>5}")

Year    | Int? | Hek. 1 | Length
--------------------------------
433/432 |  I   | Jun-28 |   384
432/431 |  O   | Jul-17 |   355
431/430 |  O   | Jul-07 |   354
--------------------------------
430/429 |  I   | Jun-26 |   384
429/428 |  O   | Jul-14 |   354
428/427 |  I   | Jul-03 |   384
427/426 |  O   | Jul-22 |   354
426/425 |  I   | Jul-11 |   384
--------------------------------
425/424 |  O   | Jul-29 |   355
424/423 |  O   | Jul-19 |   355
423/422 |  O   | Jul-09 |   354
422/421 |  I   | Jun-28 |   384
421/420 |  O   | Jul-16 |   354
--------------------------------
420/419 |  O   | Jul-05 |   354
419/418 |  I   | Jul-24 |   384
418/417 |  O   | Jul-13 |   354
417/416 |  I   | Jul-01 |   384
416/415 |  O   | Jul-20 |   355
--------------------------------
415/414 |  I   | Jul-10 |   384
414/413 |  I   | Jul-29 |   384
413/412 |  O   | Aug-16 |   354
412/411 |  O   | Aug-05 |   354
411/410 |  O   | Jul-25 |   354
--------------------------------
410/409 |  O   | Jul-14 |   355
4

How does `heniautos` compare? The output below compares `heniautos` calculations for the same years. Results that do not match are marked with an "x" (✗). The dates and year lengths indicate how far before (negative numbers) or after (positive) Meritt's results `heniautos`' results fall.

In [4]:
def check_i(m, i):
    return f"{m}  " if m == meritt_intercalations[i] else f"{m} ✗" 
    
    
def fmt_i(months, i):
    return check_i("I", i) if months > 12 else check_i("O", i)


def check_m(m, year, i):
    return f"{m} " if m == meritt_hek[i] else f"{m}✗" 
    

def fmt_m(month, year):
    mdate = ha.date(ha.bce_as_negative(year), meritt_hek1[i][0], meritt_hek1[i][1])
    date_diff = ha.span(mdate, month['days'][0]['date'])
    fmt_date = ha.as_eet(month['days'][0]['date'])[9:]
    if date_diff:
        return f"{fmt_date} ✗ ({date_diff:+3})"

    return f"{fmt_date:14}"

def fmt_len(f, i):
    ln = f[-1]['days'][-1]['doy']
    ln_diff = ln - meritt_len[i]
    
    if ln_diff:
        return f"{ln} ✗ ({ln_diff:+3})"

    return f"{ln:<11}"

for i, year in enumerate(reversed(range(405, 434))):
    fest = ha.festival_calendar(ha.bce_as_negative(year))
    #pryt = ha.prytany_calendar(ha.bce_as_negative(year))
    if not year % 5:
        print("-"*60)
    print(f"{year}/{year-1} | {fmt_i(len(fest), i)} | "
         f"{fmt_m(fest[0], year)} | {fmt_len(fest, i)}")

433/432 | I   | Jun-28         | 384        
432/431 | O   | Jul-17         | 355        
431/430 | I ✗ | Jul-07         | 383 ✗ (+29)
------------------------------------------------------------
430/429 | O ✗ | Jul-25 ✗ (+29) | 354 ✗ (-30)
429/428 | O   | Jul-13 ✗ ( -1) | 354        
428/427 | I   | Jul-02 ✗ ( -1) | 384        
427/426 | O   | Jul-21 ✗ ( -1) | 355 ✗ ( +1)
426/425 | O ✗ | Jul-11         | 355 ✗ (-29)
------------------------------------------------------------
425/424 | I ✗ | Jun-30 ✗ (-29) | 384 ✗ (+29)
424/423 | O   | Jul-19         | 354 ✗ ( -1)
423/422 | I ✗ | Jul-08 ✗ ( -1) | 384 ✗ (+30)
422/421 | O ✗ | Jul-27 ✗ (+29) | 354 ✗ (-30)
421/420 | O   | Jul-15 ✗ ( -1) | 354        
------------------------------------------------------------
420/419 | I ✗ | Jul-04 ✗ ( -1) | 384 ✗ (+30)
419/418 | O ✗ | Jul-23 ✗ ( -1) | 354 ✗ (-30)
418/417 | O   | Jul-12 ✗ ( -1) | 355 ✗ ( +1)
417/416 | I   | Jul-01         | 384        
416/415 | O   | Jul-20         | 355        
-------

For 18 of 28 years (64%), there is agreement about intercalation. The mismatches are not random but offset each other exactly so that overall agreement is good. For instance, for the years 426/5 and 425/4 Merrit makes the first intercalated  and `heniautos` the second, but both agree that the two years together ran from July 11, 426 to July 18, 424. This lack of agreement has no effect on the years before or after.

39% of the Hekatombaiṓn dates match but most of them are trivial differences of a single day. Meritt used the dates of visible new moon as calculated by Parker and Dubberstein (1942). Today we have more accurate calculations made by computer so we can expect improvement, and since we do not know how the Athenians made their observations a single day is well with expected uncertainty. Only 6 dates (21%) are different by more than a day and represent disgreements about intercalation.

The matches and mismatches in the lengths of years are another way of stating the first two categories. They total 0 for the run of 28 years! This guarantees that there is no compounding error causing the calculation to get more and more out of line with each other.

## Works Cited

* Meritt, Benjamin D. 1961. _The Athenian Year_. Sather Classical Lectures 32. Berkeley: University of California Press.
* Parker, Richard Anthony, and Waldo H. Dubberstein. 1942. _Babylonian Chronology 626 B. C.-45_. Studies in Ancient Oriental Civilization 24. Chicago: The University of Chicago Press. https://oi.uchicago.edu/research/publications/saoc/saoc-24-babylonian-chronology-626-bc-%E2%80%93-ad-45.
